### ISSUE TITLE
thetvdb_lookup issues

### ISSUE THREAD

**Issue by jasonlyle88** [Flexget/Flexget#2499]
### Expected behaviour:
To be able to get episode information from `thetvdb_lookup` plugin

### Actual behaviour:
When I dump all entries, I get `<LazyField - value will be determined when it is accessed>` for all of the tvdb entries, even the ones I have attempted to access. They entries just report a value of `None`

### Steps to reproduce:
```shell
mkdir -p "/nas/diskstation/torrents/staging/shows/"
touch "/nas/diskstation/torrents/staging/shows/[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p].mkv"
rm /opt/flexget/.flexget/{db-config.sqlite,flexget.log}
flexget execute --tasks tvdb_lookup_test --disable-tracking --dump
```

#### Config:
```yaml
tasks:
  tvdb_lookup_test:
    manual: yes
    assume_quality:
      720p: hdtv
      1080p: hdtv
      hdtv: 480p
    filesystem:
      path: "/nas/diskstation/torrents/staging/shows/"
      regexp: ".*HorribleSubs.*.mkv"
    accept_all: yes
    series:
      - "Boruto - Naruto Next Generations (2017)":
          identified_by: sequence
          set:
            tvdb_id: 321285
    thetvdb_lookup: yes
    exec:
      on_output:
        # for_accepted: printf '%s' '{{tvdb_series_name}}'
        for_accepted: printf '%s - %s' '{{tvdb_series_name}}' '{{tvdb_absolute_number}}'
```
  
#### Log:
With `for_accepted: printf '%s' '{{tvdb_series_name}}'`
<details>
 <summary>(click to expand)</summary>

```
2019-11-16 14:04 VERBOSE  manager                       Creating new database /opt/flexget/.flexget/db-config.sqlite - DO NOT INTERRUPT ...
2019-11-16 14:04 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2019-11-16 14:04 VERBOSE  assume_quality tvdb_lookup_test New assumption: 720p is hdtv
2019-11-16 14:04 VERBOSE  assume_quality tvdb_lookup_test New assumption: 1080p is hdtv
2019-11-16 14:04 VERBOSE  assume_quality tvdb_lookup_test New assumption: hdtv is 480p
2019-11-16 14:04 VERBOSE  filesystem    tvdb_lookup_test Starting to scan folders.
2019-11-16 14:04 VERBOSE  filesystem    tvdb_lookup_test Scanning folder /nas/diskstation/torrents/staging/shows/. Recursion is set to False.
2019-11-16 14:04 VERBOSE  details       tvdb_lookup_test Produced 1 entries.
2019-11-16 14:04 VERBOSE  assume_quality tvdb_lookup_test [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
2019-11-16 14:04 VERBOSE  assume_quality tvdb_lookup_test New quality: 1080p hdtv
2019-11-16 14:04 VERBOSE  task          tvdb_lookup_test ACCEPTED: `[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]` by accept_all plugin
2019-11-16 14:04 VERBOSE  details       tvdb_lookup_test Summary - Accepted: 1 (Rejected: 0 Undecided: 0 Failed: 0)
2019-11-16 14:04 VERBOSE  exec          tvdb_lookup_test Executing: printf '%s' 'None'
2019-11-16 14:04 INFO     exec          tvdb_lookup_test Stdout: None
-- Accepted: ---------------------------
title            : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
url              : file:///nas/diskstation/torrents/staging/shows/%5BHorribleSubs%5D%20Boruto%20-%20Naruto%20Next%20Generations%20-%20132%20%5B1080p%5D.mkv
original_url     : file:///nas/diskstation/torrents/staging/shows/%5BHorribleSubs%5D%20Boruto%20-%20Naruto%20Next%20Generations%20-%20132%20%5B1080p%5D.mkv
accepted_by      : accept_all
accessed         : 2019-11-15 17:23:55.140467
assumed_quality  : True
content_size     : 0
created          : 2019-11-15 17:23:29.280467
filename         : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p].mkv
id               : boruto - naruto next generations (2017) 132
imdb_id          : <LazyField - value will be determined when it is accessed>
imdb_url         : <LazyField - value will be determined when it is accessed>
location         : /nas/diskstation/torrents/staging/shows/[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p].mkv
modified         : 2019-11-15 10:16:19.168042
original_title   : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
proper           : False
proper_count     : 0
quality          : 1080p hdtv
release_group    : None
season_pack      : None
series_episodes  : 1
series_exact     : True
series_id        : 132
series_id_type   : sequence
series_identified_by: sequence
series_name      : Boruto - Naruto Next Generations (2017)
series_parser    : <SeriesParseResult(data=[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p],name=Boruto - Naruto Next Generations (2017),id=132,season=0,season_pack=None,episode=132,quality=1080p,proper=0,special=False,status=OK)>
series_releases  : [1]
series_season    : 2019
task             : tvdb_lookup_test
timestamp        : 2019-11-15 10:16:19.168042
tvdb_absolute_number: <LazyField - value will be determined when it is accessed>
tvdb_actors      : <LazyField - value will be determined when it is accessed>
tvdb_air_time    : <LazyField - value will be determined when it is accessed>
tvdb_airs_day_of_week: <LazyField - value will be determined when it is accessed>
tvdb_banner      : <LazyField - value will be determined when it is accessed>
tvdb_content_rating: <LazyField - value will be determined when it is accessed>
tvdb_ep_air_date : <LazyField - value will be determined when it is accessed>
tvdb_ep_directors: <LazyField - value will be determined when it is accessed>
tvdb_ep_id       : <LazyField - value will be determined when it is accessed>
tvdb_ep_image    : <LazyField - value will be determined when it is accessed>
tvdb_ep_name     : <LazyField - value will be determined when it is accessed>
tvdb_ep_overview : <LazyField - value will be determined when it is accessed>
tvdb_ep_rating   : <LazyField - value will be determined when it is accessed>
tvdb_episode     : <LazyField - value will be determined when it is accessed>
tvdb_first_air_date: <LazyField - value will be determined when it is accessed>
tvdb_genres      : <LazyField - value will be determined when it is accessed>
tvdb_id          : 321285
tvdb_language    : <LazyField - value will be determined when it is accessed>
tvdb_network     : <LazyField - value will be determined when it is accessed>
tvdb_overview    : <LazyField - value will be determined when it is accessed>
tvdb_posters     : <LazyField - value will be determined when it is accessed>
tvdb_rating      : <LazyField - value will be determined when it is accessed>
tvdb_runtime     : <LazyField - value will be determined when it is accessed>
tvdb_season      : <LazyField - value will be determined when it is accessed>
tvdb_series_name : <LazyField - value will be determined when it is accessed>
tvdb_status      : <LazyField - value will be determined when it is accessed>
tvdb_url         : <LazyField - value will be determined when it is accessed>
zap2it_id        : <LazyField - value will be determined when it is accessed>
```
</details>

With `for_accepted: printf '%s - %s' '{{tvdb_series_name}}' '{{tvdb_absolute_number}}' `
<details>
 <summary>(click to expand)</summary>

```
2019-11-16 14:05 VERBOSE  manager                       Creating new database /opt/flexget/.flexget/db-config.sqlite - DO NOT INTERRUPT ...
2019-11-16 14:05 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2019-11-16 14:05 VERBOSE  assume_quality tvdb_lookup_test New assumption: 720p is hdtv
2019-11-16 14:05 VERBOSE  assume_quality tvdb_lookup_test New assumption: 1080p is hdtv
2019-11-16 14:05 VERBOSE  assume_quality tvdb_lookup_test New assumption: hdtv is 480p
2019-11-16 14:05 VERBOSE  filesystem    tvdb_lookup_test Starting to scan folders.
2019-11-16 14:05 VERBOSE  filesystem    tvdb_lookup_test Scanning folder /nas/diskstation/torrents/staging/shows/. Recursion is set to False.
2019-11-16 14:05 VERBOSE  details       tvdb_lookup_test Produced 1 entries.
2019-11-16 14:05 VERBOSE  assume_quality tvdb_lookup_test [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
2019-11-16 14:05 VERBOSE  assume_quality tvdb_lookup_test New quality: 1080p hdtv
2019-11-16 14:05 VERBOSE  task          tvdb_lookup_test ACCEPTED: `[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]` by accept_all plugin
2019-11-16 14:05 VERBOSE  details       tvdb_lookup_test Summary - Accepted: 1 (Rejected: 0 Undecided: 0 Failed: 0)
2019-11-16 14:05 VERBOSE  exec          tvdb_lookup_test Executing: printf '%s - %s' 'None' 'None'
2019-11-16 14:05 INFO     exec          tvdb_lookup_test Stdout: None - None
-- Accepted: ---------------------------
title            : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
url              : file:///nas/diskstation/torrents/staging/shows/%5BHorribleSubs%5D%20Boruto%20-%20Naruto%20Next%20Generations%20-%20132%20%5B1080p%5D.mkv
original_url     : file:///nas/diskstation/torrents/staging/shows/%5BHorribleSubs%5D%20Boruto%20-%20Naruto%20Next%20Generations%20-%20132%20%5B1080p%5D.mkv
accepted_by      : accept_all
accessed         : 2019-11-15 17:23:55.140467
assumed_quality  : True
content_size     : 0
created          : 2019-11-15 17:23:29.280467
filename         : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p].mkv
id               : boruto - naruto next generations (2017) 132
imdb_id          : <LazyField - value will be determined when it is accessed>
imdb_url         : <LazyField - value will be determined when it is accessed>
location         : /nas/diskstation/torrents/staging/shows/[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p].mkv
modified         : 2019-11-15 10:16:19.168042
original_title   : [HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p]
proper           : False
proper_count     : 0
quality          : 1080p hdtv
release_group    : None
season_pack      : None
series_episodes  : 1
series_exact     : True
series_id        : 132
series_id_type   : sequence
series_identified_by: sequence
series_name      : Boruto - Naruto Next Generations (2017)
series_parser    : <SeriesParseResult(data=[HorribleSubs] Boruto - Naruto Next Generations - 132 [1080p],name=Boruto - Naruto Next Generations (2017),id=132,season=0,season_pack=None,episode=132,quality=1080p,proper=0,special=False,status=OK)>
series_releases  : [1]
series_season    : 2019
task             : tvdb_lookup_test
timestamp        : 2019-11-15 10:16:19.168042
tvdb_absolute_number: <LazyField - value will be determined when it is accessed>
tvdb_actors      : <LazyField - value will be determined when it is accessed>
tvdb_air_time    : <LazyField - value will be determined when it is accessed>
tvdb_airs_day_of_week: <LazyField - value will be determined when it is accessed>
tvdb_banner      : <LazyField - value will be determined when it is accessed>
tvdb_content_rating: <LazyField - value will be determined when it is accessed>
tvdb_ep_air_date : <LazyField - value will be determined when it is accessed>
tvdb_ep_directors: <LazyField - value will be determined when it is accessed>
tvdb_ep_id       : <LazyField - value will be determined when it is accessed>
tvdb_ep_image    : <LazyField - value will be determined when it is accessed>
tvdb_ep_name     : <LazyField - value will be determined when it is accessed>
tvdb_ep_overview : <LazyField - value will be determined when it is accessed>
tvdb_ep_rating   : <LazyField - value will be determined when it is accessed>
tvdb_episode     : <LazyField - value will be determined when it is accessed>
tvdb_first_air_date: <LazyField - value will be determined when it is accessed>
tvdb_genres      : <LazyField - value will be determined when it is accessed>
tvdb_id          : 321285
tvdb_language    : <LazyField - value will be determined when it is accessed>
tvdb_network     : <LazyField - value will be determined when it is accessed>
tvdb_overview    : <LazyField - value will be determined when it is accessed>
tvdb_posters     : <LazyField - value will be determined when it is accessed>
tvdb_rating      : <LazyField - value will be determined when it is accessed>
tvdb_runtime     : <LazyField - value will be determined when it is accessed>
tvdb_season      : <LazyField - value will be determined when it is accessed>
tvdb_series_name : <LazyField - value will be determined when it is accessed>
tvdb_status      : <LazyField - value will be determined when it is accessed>
tvdb_url         : <LazyField - value will be determined when it is accessed>
zap2it_id        : <LazyField - value will be determined when it is accessed>
```
</details>

### Additional information:

- FlexGet version: 3.0.4
- Python version: 3.7.5rc1
- Installation method: pip
- Using daemon (yes/no): no
- OS and version: Ubuntu 19.10

**Comment by BrutuZ** [comment]
Likely related to TVDB API migration causing a whole lot of issues all around https://github.com/Flexget/Flexget/issues/2498#issuecomment-554588482

**Comment by jasonlyle88** [comment]
It looks like thetvdb did switch over to the new API version, but from what I understand of the documentation, v2 of the API is still accessible when you specifically request it. 

If I may, I would propose that (until someone has the time to update everything for the new 3.0 API) the old version of the API be specifically requested by flexget's thetvdb code as mentioned [here](https://api.thetvdb.com/swagger#versioning).

It sounds like thetvdb is having some of their own issues with the cutover, but I believe this might be a great "band-aid" until V3 is ready and stable on both sides.

**Comment by BrutuZ** [comment]
Not the case of version incompatibility. Didn't dig much into the details but their schema doesn't seem to have changed much, it should be mostly retro-compatible from what I could gather.

I tested this specific example you gave and the problem is with the API not returning data, not FlexGet not knowing how to handle it. In fact, I tested a couple other show IDs (both via curl and their swagger page) and they all returned the same error. 
```
$ curl -X GET --header "Accept: application/json" --header "Authorization: Bearer %jwd%" "https://api.thetvdb.com/series/321285"
{"Error":"ID: 321285 not found"}
```


They still have some tweaking to do on V3 to even out these migration bugs, not much we can do but wait until then.

**Comment by jasonlyle88** [comment]
Well, that would make it rather difficult to do anything. Thanks for the investigation! As this isn't an issue on flexget's side, I'm going to go ahead and close this issue.

**Comment by jasonlyle88** [comment]
Looking into thetvdb API, the episodes endpoints are still working (even thought the series is not). Maybe this is worth looking in to: I would assume that the plugin could (possibly even should) be refactored so that the episode and series information can be looked up regardless of issues in one or the other.

**Comment by zmasterar** [comment]
I just tested the API, and the series endpoint is working for me

```
curl -X GET --header 'Accept: application/json' --header 'Authorization: Bearer token' 'https://api.thetvdb.com/series/321285'

{
  "data": {
    "id": 321285,
    "seriesId": "",
    "seriesName": "Boruto: Naruto Next Generations",
    "aliases": [],
    "season": "1",
    "poster": "posters/5da1013c31647.jpg",
    "banner": "graphical/5c429e990b029.jpg",
    "fanart": "fanart/original/5d76c9470d4d1.jpg",
    "status": "Continuing",
    "firstAired": "2015-08-07",
    "network": "TV Tokyo",
    "networkId": "285",
    "runtime": "25",
    "language": "en",
    "genre": [
      "Action",
      "Adventure",
      "Animation",
      "Anime"
    ],
    "overview": "The Hidden Leaf Village has entered an era of peace and modernity. Tall buildings line the streets, giant screens flash with images, and the Thunder Rail runs through the village, connecting each district together. Though it's still a ninja village, the number of civilians has increased and the life of the shinobi is beginning to change.\r\n \r\nBoruto Uzumaki, son of Seventh Hokage Naruto Uzumaki, has enrolled in the Ninja Academy to learn the ways of the ninja. The other students are ready to dismiss him as \"just the son of the Hokage,” but Boruto’s heart and character blow all their assumptions away.\r\n \r\nAs a series of mysterious events begins to unfold, it’s up to Boruto and his new friends to handle them. Like a gale-force wind, Boruto makes his own way into everyone's hearts; his story is about to begin!!\r\n",
    "lastUpdated": 1573829355,
    "airsDayOfWeek": "Sunday",
    "airsTime": "5:30 PM",
    "rating": null,
    "imdbId": "tt6342474",
    "zap2itId": "",
    "added": "2016-12-17 07:59:09",
    "addedBy": 128397,
    "siteRating": 0,
    "siteRatingCount": 0,
    "slug": "boruto-naruto-next-generations"
  }
}
```
but still flexget the tvdb entries data is not populating

**Comment by jasonlyle88** [comment]
I am getting data back from the series and episodes endpoints now (`/series/{id}` and `series/321285/episodes/query`). However Only the series information is getting populated in flexget. As such I am re-opening this issue to see what is going on with the cetain data not being populated in the flexget plugin.

**Comment by frankyw** [comment]
This is probably due to a still broken tvdb API, other apps like [emby](https://emby.media/community/index.php?/topic/79269-tvdb-api-issues/page-8#entry808164) are also affected (in terms of episode data failing)

**Comment by jasonlyle88** [comment]
@frankyw The point of my previous post is that the TVDB API is returning that data now. However, it is still not coming through on flexget. I have run the API calls via postman and CURL and both return the expected data. Flexget, however, is not.

**Comment by jasonlyle88** [comment]
Not sure if the issue ended up being with the API or with flexget. I upgraded to 3.0.7 and this is working now, so I will go ahead and close the issue.

**Comment by BrutuZ** [comment]
Likely fixed by #2505 since the fail fetching that field (`director`) aborted the whole entry to be populated.

**Comment by jasonlyle88** [comment]
Most likely. Thanks for the work @BrutuZ!
